### PR TITLE
Sync team membership & permissions to GitHub

### PR DESCRIPTION
**Pull request by stefanv**
It looks for `teams.yaml` in the current directory, formatted as follows:

```yaml

- name: SPEC Steering Committee
  description:
  members:
    - stefanv
    - ivanov
    ...
  permissions:
    - repo: specs
      role: maintain

- name: Community Managers
  description: Scientific Python Community Managers
  members:
    - stefanv
    - jarrodmillman
    ...
  permissions:
    - repo: specs
      role: triage
```

It requires a classic token with permissions "Repo" and "Admin".

### CODE REVIEW

**Comment by stefanv** [comment]
Example `teams.yaml`:

<details>

```
# Permissions can be: read, triage, write, maintain, admin
# Repo roles can be: pull, triage, push, maintain, admin

- name: SPEC Steering Committee
  description:
  members:
    - stefanv
    - rgommers
    - ivanov
    - jarrodmillman
    - Carreau
    - jni
    - jorisvandenbossche
    - rossbar
    - pllim
    - kthyng
    - MridulS
    - munkm
    - bsipocz
    - GeorgianaElena
    - andersy005
    - juanis2112
    - lagru
    - MSanKeys963
    - tupui
    - kne42
    - InessaPawson
  permissions:
    - repo: specs
      role: maintain

- name: Community Managers
  description: Scientific Python Community Managers
  members:
    - stefanv
    - jarrodmillman
    - rossbar
    - mkcor
    - melissawm
    - davzaman
    - MridulS
    - crazy4pi314
    - bsipocz
    - juanis2112
    - tupui
    - kne42
  permissions:
    - repo: specs
      role: triage

- name: Test Team
  description: This is the description of test team
  members:
    - stefanv
    - jarrodmillman
  permissions:
    - repo: scientific-python.org
      role: maintain
```

</details>

TODO:

- Better printing of what is happening (adding team member X, removing team member Y, setting repo permission, etc.)
- Handle subteams

Features I chose not to implement:

- Removing teams

**Comment by stefanv** [comment]
Re: the config file, it only tries to sync teams mentioned in the config. I.e., if you describe a subset of teams, it will leave the rest of the teams alone.

**Review by jarrodmillman — APPROVED** [review]

**Comment by bsipocz** [comment]
> - name: SPEC Steering Committee
  description:
  members:
    - stefanv
    - ivanov
    ...


It's nitpicking at this point, but could we make the members sections alphabetical? That way it would be easier to spot duplicates.

**Comment by stefanv** [comment]
@bsipocz Sure, we can do that in the members repo, where this data will be stored.

File: sync-teams-to-gh.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+            return data
+
+        if "message" in data:
+            qprint(f"Error retrieving {url}: {data['message']}")
+            sys.exit(1)
+
+        if "next" in r.links:
+            page += 1
+        else:
+            more_pages = False
```
Comment (stefanv) [inline sync-teams-to-gh.py:87]: Paged data should be appended to the returned result here.

File: sync-teams-to-gh.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
```
Comment (jarrodmillman) [inline sync-teams-to-gh.py:88]:
```suggestion
    teams = []
    while more_pages:
        if page == 1:
            qprint(f'{DARK_GRAY}GET {url}{RESET}')
        else:
            qprint(f'{DARK_GRAY}GET {url} [{page}]{RESET}')

        r = requests.get(url + f"?page={page}", headers=headers)

        try:
            data = r.json()
        except json.decoder.JSONDecodeError:
            qprint("Error: cannot decode JSON response")
            sys.exit(1)

        if fail_ok and "message" in data:
            data["status"] = r.status_code
            return data

        if "message" in data:
            qprint(f"Error retrieving {url}: {data['message']}")
            sys.exit(1)

        if "next" in r.links:
            page += 1
        else:
            more_pages = False
            
         teams.extend(data)
```

**Pull request merged**